nixos/modules/services/monitoring/prometheus/exporters/pve.nix

```diff
@@ -102,8 +122,10 @@ in
           --${optionalString (!cfg.collectors.cluster) "no-"}collector.cluster \
           --${optionalString (!cfg.collectors.resources) "no-"}collector.resources \
           --${optionalString (!cfg.collectors.config) "no-"}collector.config \
-          %d/configFile \
-          ${toString cfg.port} ${cfg.listenAddress}
+          ${optionalString (cfg.server.keyFile != null) "--server.keyfile ${cfg.server.keyFile} \\"}
+          ${optionalString (cfg.server.certFile != null) "--server.certfile ${cfg.server.certFile} \\"}
```

Comment:
This will make the service fail unless both `keyFile` and `certFile` are set. This is what the generated unit file contains:
```
$ cat /nix/store/i93h5injlwzz5n2wzzycgz0lp31ldp59-unit-prometheus-pve-exporter.service/prometheus-pve-exporter.service
[Unit]
After=network.target

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/asw7r5sg4gavghq263106ngjqkwd8yid-glibc-locales-2.38-27/lib/locale/locale-archive"
Environment="PATH=/nix/store/khndnv11g1rmzhzymm1s5dw7l2ld45bc-coreutils-9.4/bin:/nix/store/7iyzsc224ax9dm71fv4grndixylcn7kh-findutils-4.9.0/bin:/nix/store/mmsb0ivm355r4l3yjbpaiirkf673n66v-gnugrep-3.11/bin:/nix/store/nbsz4b0kdh5a2b9h2cr9ngxskfqx575k-gnused-4.9/bin:/nix/store/vfmf8qh892jfl107hih0yfnic00byjgj-systemd-254.6/bin:/nix/store/khndnv11g1rmzhzymm1s5dw7l2ld45bc-coreutils-9.4/sbin:/nix/store/7iyzsc224ax9dm71fv4grndixylcn7kh-findutils-4.9.0/sbin:/nix/store/mmsb0ivm355r4l3yjbpaiirkf673n66v-gnugrep-3.11/sbin:/nix/store/nbsz4b0kdh5a2b9h2cr9ngxskfqx575k-gnused-4.9/sbin:/nix/store/vfmf8qh892jfl107hih0yfnic00byjgj-systemd-254.6/sbin"
Environment="TZDIR=/nix/store/if62v5igsm4zjzag9zb2xldmzfg8wj3d-tzdata-2023d/share/zoneinfo"
CapabilityBoundingSet=
DeviceAllow=
DynamicUser=true
ExecStart=/nix/store/f9kyb6bri2iangimjllh5c0ydfdnndix-prometheus-pve-exporter-3.2.1/bin/pve_exporter \
  --collector.status \
  --collector.version \
  --collector.node \
  --collector.cluster \
  --collector.resources \
  --collector.config \


  --config.file %d/configFile \
  --web.listen-address 0.0.0.0:9221

Group=pve-exporter
LoadCredential=configFile:/nix/store/sfq3awx4x3y85lzlb97zra6pfc3baz1m-pve.yml
LockPersonality=true
MemoryDenyWriteExecute=true
NoNewPrivileges=true
PrivateDevices=true
PrivateTmp=true
ProtectClock=true
ProtectControlGroups=true
ProtectHome=true
ProtectHostname=true
ProtectKernelLogs=true
ProtectKernelModules=true
ProtectKernelTunables=true
ProtectSystem=strict
RemoveIPC=true
Restart=always
RestrictAddressFamilies=AF_INET
RestrictAddressFamilies=AF_INET6
RestrictNamespaces=true
RestrictRealtime=true
RestrictSUIDSGID=true
SystemCallArchitectures=native
UMask=0077
User=pve-exporter
WorkingDirectory=/tmp

[Install]
WantedBy=multi-user.target
```

The unescaped newlines in between will cause it to break. The backslashes need to be moved outside the `optionalString`.

Reply:
Fixed